typeshed_client/parser.py

```diff
@@ -430,7 +430,17 @@ def generic_visit(self, node: ast.AST) -> Iterable[NameInfo]:
         raise InvalidStub(f"Cannot handle node {ast.dump(node)}", self.file_path)
 
 
-class _LiteralEvalVisitor(ast.NodeVisitor):
+class LiteralEvalVisitor(ast.NodeVisitor):
+    """Visitor to evaluate the truthiness of a ``test`` expression in an ``ast.Compare`` node.
+
+    ``LiteralEvalVisitor(ctx, path).visit(node)`` will return ``True`` if ``node`` is an
+    expression that can be statically determined to always be ``True``, ``False`` if it can
+    be statically determined to always be ``False``, and ``None`` if its truthiness cannot
```

Comment:
Actually I think this isn't true, it might also return some arbitrary other non-bool object (see e.g. the `visit_Tuple` method below). In typeshed-client's own use we just treat whatever it returns as a boolean-like object, so typeshed-client would consider `if ("some", "tuple"):` to be statically true.

Reply:
Yes, `_visit_condition` does only return True/False/None, but it may also raise InvalidStub.

It also has branching on whether or not we're in a `.py` file. That's for cases where a library has an `__init__.py` at the root and stubs elsewhere (concretely, pytorch: #117).

Reply:
I think making something like `visit_condition(node, ctx, path, is_py_file=False)` a top-level global function should work well.

Reply:
cool, I'll make a PR to fix this up.